src/metatron.rs

```diff
@@ -0,0 +1,135 @@
+use super::*;
+
+pub(crate) struct Metatron {
+    blocks: AtomicU64,
+    shares: AtomicU64,
+    started: Instant,
+    workers: AtomicU64,
+}
+
+impl Metatron {
+    pub(crate) fn new() -> Self {
+        Self {
+            blocks: AtomicU64::new(0),
+            shares: AtomicU64::new(0),
+            started: Instant::now(),
+            workers: AtomicU64::new(0),
+        }
+    }
+
+    pub(crate) fn add_block(&self) {
+        self.blocks.fetch_add(1, Ordering::Relaxed);
+    }
+
+    pub(crate) fn add_share(&self) {
+        self.shares.fetch_add(1, Ordering::Relaxed);
+    }
+
+    pub(crate) fn add_worker(&self) {
+        self.workers.fetch_add(1, Ordering::Relaxed);
+    }
+
+    pub(crate) fn sub_worker(&self) {
+        self.workers.fetch_sub(1, Ordering::Relaxed);
+    }
```

Comment:
The current implementation correctly pairs `add_worker` in constructor with `sub_worker` in `Drop`, however, if `sub_worker()` is called when the count is 0, it wraps to `u64::MAX`. It will only occur due to bugs in our calling code/connection lifecycle. 

This is lower severity, it's just going to make our code a bit more defensive.
```suggestion
    pub(crate) fn sub_worker(&self) {
        let previous = self.workers.fetch_sub(1, Ordering::Relaxed);
        debug_assert!(previous > 0, "worker count underflow - mismatched add/sub calls");
    }
```